pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1094,6 +1050,53 @@ func (c *K8sOrchestrator) GetAllK8sVolumes() []string {
 	return volumeIDs
 }
 
+// SetWcpCapabilitiesMap reads the capabilities values from 'supervisor-capabilities' CR in
+// supervisor cluster and sets values in global wcp capabilities map.
+func SetWcpCapabilitiesMap(ctx context.Context, cnsOperatorClient client.Client) error {
+	log := logger.GetLogger(ctx)
+	wcpCapabilitiesMutex.Lock()
```

Comment:
Why do we need a lock here, is the capability CR modified/updated by other components on regular basis ? 
I was under the impression that the CR is modified only during supervisor upgrade, and csi driver is part of the wcp package. Also, I see that the method is called in a ticker which I every 2 minutes. 
I am not sure why we want to lock the CR every minutes, can you add some explanation

Reply:
This lock is for wcpCapabilities global Map and not for CR. I have renamed the lock name to avoid confusion and moved it to appropriate place. We don't need this lock while fetching data from CR.